modules/tests/it/src/test/scala/com/github/scoquelin/arugula/BaseRedisCommandsIntegrationSpec.scala

```diff
@@ -57,12 +56,6 @@ trait BaseRedisCommandsIntegrationSpec extends AsyncWordSpecLike with TestContai
     cachedClients = RedisCommandsCachedClients(redisSingleNodeConfig, redisClusterConfig)
   }
 
-  override def afterEach(): Unit = {
-    //flushing both redis instances after each test
-    cachedClients.getClient(RedisCodec.Utf8WithValueAsStringCodec, SingleNode).flushAll
-    cachedClients.getClient(RedisCodec.Utf8WithValueAsStringCodec, Cluster).flushAll
```

Comment:
these flush commands are futures and nothing is blocking to wait for them to complete. That makes the tests non deterministic since new commands could be issued by the next test before this flush command happens.

Reply:
Yeah good catch! We should make them blocking and/or find a different strategy to clean up in between tests. Did you experience failing tests related to that issue ?

Reply:
As long as we use the random key approach rather than reusing the same hard-coded strings it mostly all works out and we just clean up at the end. The places where it doesn't work as well are places where there are scans but in those cases we can use a key prefix pattern and it all works fine. 
I experimented with flushing with Await but it really slows down the test runs and I don't think it is necessary. A little care in namespacing keys and using random key names is sufficient.